site/source/tutorials/1010_containers_and_injecting_state/index.md

```diff
@@ -0,0 +1,175 @@
+---
+layout: tutorial
+title: State management
+overview: In this tutorial, you will learn what Containers and Injectors are and how to use them to coordinate external state and inject as properties into the sections of the widget tree.
+---
+
+{% section 'first' %}
+
+# State management
+
+## Overview
+
+Modern web application today are required to manage complex state models which can involve fetching data from a remote service or multiple widget requiring the same slices of state. While Dojo 2's widgets can manage application state, encapsulation and a clean separation of concerns may be lost if widgets manage their own visual representations, listen for interactions from the user, manage their children, and keep track of state information. Additionally, using widgets to pass state through an application often forces the widgets to be aware of state information for the sole purpose of passing that data down to their children. To allow widgets to remain focused on their primary roles of providing a visual representation of themselves and listening for user interactions, Dojo 2 provides two classes, `Container`, and `Injector` that are designed to coordinate an application's external state and connect and map this state to properties only for the widgets that it is required.
+
+In this tutorial, we will start with an application that is managing its state in the widgets themselves. We will then extract all of the state-related code out of the widgets and inject external state as properties only into widgets as is needed. You can [download](../assets/1010_form_widgets-initial.zip) the demo project to get started.
+
+## Prerequisites
+
+This tutorial assumes that you have gone through the [beginner tutorial](../001_static_content) series.
+
+{% section %}
+
+## Creating an application context
+
+{% task 'Create a class to manage application state.' %}
+
+To begin our discussion, let's review the initial version of `App`:
+
+{% include_codefile 'demo/initial/biz-e-corp/src/widgets/App.ts' %}
+
+Most of this widget is dedicated to holding and managing the `WorkerData` in the application. Notice, however, that it never actually uses that data itself. `App` is only containing the state and passing it to the children as required via properties. Up lifting state to the highest common widget in the tree is a valid pattern but as an application's state grows in size and complexity it is often desirable to decouple this from widgets. In larger applications, the `App` class would become complicated and more difficult to maintain due to the additional state information that it would be required track. Since the state information is not a primary concern of the `App` class, let's refactor it out of `App` and into a new `ApplicationContext` class.
```

Comment:
"Up lifting state" -> should be "Uplifting state", though perhaps elevating or raising is a better choice than uplifting?

Reply:
"pattern but as an application's state grows in size and complexity it is often" -> "pattern, but as an application's state grows in size and complexity, it is often"

Reply:
"required track" -> "required to track"

Reply:
Change to "lifting up state", it is a pattern that is familiar in the community (ala React)